src/shared_data.cc

```diff
@@ -25,7 +25,13 @@
 
 namespace proxy_wasm {
 
-SharedData global_shared_data;
+std::mutex global_shared_data_mutex;
+
+SharedData *getGlobalSharedData() {
+  std::lock_guard<std::mutex> lock(global_shared_data_mutex);
+  static auto *ptr = new SharedData;
+  return ptr;
+};
```

Comment:
Why do we need this mutex here? Also, it gets out of scope, so access to the returned object won't be guarded.

Reply:
I just misunderstand that static local initialization is not thread safe. will fix this